utils/parallel.go

```diff
@@ -164,3 +164,49 @@ func RunInParallel(ctx context.Context, fs []SimpleFunc) (time.Duration, error)
 	wg.Wait()
 	return time.Since(start), bigError
 }
+
+// FloatFunc is for GetInParallel.
+type FloatFunc func(ctx context.Context) (float64, error)
+
+// GetInParallel runs all functions in parallel, return is elapsed time and n error.
+func GetInParallel(ctx context.Context, fs []FloatFunc) (time.Duration, []float64, error) {
+	start := time.Now()
+	ctx, cancel := context.WithCancel(ctx)
+	var floats []float64
+
+	var wg sync.WaitGroup
+
+	var bigError error
+	var bigErrorMutex sync.Mutex
+	storeError := func(float float64, err error) {
+		bigErrorMutex.Lock()
+		defer bigErrorMutex.Unlock()
+		if bigError == nil || !errors.Is(err, context.Canceled) {
```

Comment:
similarly here (which you probably now realize) this was copied from the other code. seems to me like this is written this way because we just don't want to include the context canceled error. maybe because we want to continue running the other functions even if one function is done and the context is canceled? not 100% sure

Reply:
Sigh. I can't tell if you've just made a second copy of a bug or not. I wish the original version had included comments explaining this, and I wish the code were easy to extend so you could avoid copy-and-paste. but neither of those is exactly your responsibility.

I guess you can leave as-is for consistency, but I don't like it. :confused:

Reply:
if you can figure out a good way to extend the first function I am open to suggestions. but as of right now `RunInParallel` takes in a list of functions that only return errors and I specifically need to use functions that return a float to get the positions of the motors so that makes it a little difficult.

Reply:
eh... not thoughts I'm very happy with. You could make the one that doesn't return anything just call this one:
```go
func RunInParallel(ctx context.Context, fs []SimpleFunc) (time.Duration, error) {
    tempFs := make([]FloatFunc, len(fs))
    for i, f := range fs {
        tempFs[i] = func(ctx context.Context) (float64, error) {
            err := f()
            return 0, err
        }
    }
    duration, zeros, err := RunInParallel(ctx, tempFs)
    return duration, err
}
```
but that's very much a hack.

I guess just leave as-is. Sorry, I didn't realize you were waiting on me for this.

Reply:
Right, but then there's the other cases that use just regular run in parallel with no returns? like I think they are just different enough uses like a setter and a getter that it seems okay for them to be separate imo. so I think I am going to leave as-is but if it comes up again with something else we can maybe reevaluate!

Reply:
But thanks for taking another look just wanted to make sure